ibis/expr/types.py

```diff
@@ -1039,6 +1039,54 @@ def type(self):
         return dt.int64
 
 
+class UInt8Value(IntegerValue):
+
+    _implicit_casts = set([
+        'uint8', 'uint16', 'uint32', 'uint64',
+        'float', 'double', 'decimal'
+    ])
+
+    def type(self):
+        return dt.uint8
+
+
+class UInt16Value(IntegerValue):
+
+    _implicit_casts = set(['uint32', 'uint64',
+                           'float', 'double', 'decimal'])
+
+    def type(self):
+        return dt.uint16
+
+
+class UInt32Value(IntegerValue):
+
+    _implicit_casts = set(['uint64', 'float', 'double', 'decimal'])
+
+    def type(self):
+        return dt.uint32
+
+
+class UInt64Value(IntegerValue):
+
+    _implicit_casts = set(['float', 'double', 'decimal'])
+
+    def type(self):
+        return dt.uint64
+
+
+class HalffloatingValue(NumericValue):
+    pass
+
+
+class HalffloatValue(HalffloatingValue):
```

Comment:
Can we call this `HalfFloatValue`? That double `ff` is a little hard on the eyes.

Reply:
done

Reply:
Same as above, this is still here.